ddtrace/internal/writer.py

```diff
@@ -103,57 +203,27 @@ def write(self, spans=None, services=None):
                 if self._started is False:
                     self.start()
                     self._started = True
-        if spans:
-            self._trace_queue.put(spans)
+        if not spans:
+            return
 
-    def flush_queue(self):
-        traces = self._trace_queue.get()
+        encoded = self._encoder.encode_trace(spans)
+        if len(encoded) > self._max_payload_size:
+            # Trace is bigger than the payload limit, nothing we can do.
+            log.warning("trace larger than maximum payload limit, dropping")
+            return
+
+        self._buffer.put(encoded)
```

Comment:
we'll need to handle when we downgrade, how do we downgrade all traces already in the buffer?

Reply:
this is also broken currently in the trace client and we haven't heard any complaints. I wonder if it's worth dropping older agent support 🤔

Reply:
I am confident in us dropping v0.2 support, not sure about v0.3, might need to grab some data on that one, or figure out messaging

Reply:
I think v0.2 is the only problem. v0.3 supports msgpack so no need to reencode

Reply:
removed v0.2 and added an integration test for downgrading to v0.3